### DISCUSSION TITLE
DHCP Option 6 Question

### DISCUSSION THREAD

**Discussion by WildByDesign** [AdguardTeam/AdGuardHome#3298] [Q&A] [answered]
- If we use DHCP Option 6 to push DNS servers out to clients, does that bypass AdGuard Home’s ad filtering and ability to collect statistics for those users?

Thank you.

EDIT: I have another question related to DHCP Option 6. 

- Is it possible to push DHCP Option 6 out to only specific users?

For example, let’s say there are 12 devices managed by AGH’s DHCP server all setup in the Client section managed by MAC addresses. I would like 10 of those devices to continue reaching device IP for DNS lookups, while the other 2 devices have DNS servers pushed to them via DHCP Option 6.

- Is the above example possible currently in AGH?

**Comment by ainar-g** [comment, marked as answer]
Hello and sorry about the late answer.

> If we use DHCP Option 6 to push DNS servers out to clients, does that bypass AdGuard Home’s ad filtering and ability to collect statistics for those users?

Yes, unless the DHCP clients override your DNS settings and make their machines point at AGH instead, but that kind of defeats the purpose.

> Is it possible to push DHCP Option 6 out to only specific users?

Currently, no.  If you would like to have this feature, please fill a separate issue, and we will see if we can do it.  But I must warn you that that will probably not be soon.

If you want some clients to not be filtered, you can still do that without touching Option 6 in the “Client settings” section of AGH's settings.

**Comment by WildByDesign** [comment]
Thank you, sir. That answered my questions perfectly and gave me a better understanding.
